### ISSUE TITLE
Error: connect EADDRINUSE 13.**.**.152:4*3

### ISSUE THREAD

**Issue by BhargavJanardhan** [microsoft/BotFramework-DirectLineJS#299]
Hi,
**Issue**: The error message"connect EADDRINUSE 13.1*.2*.152:4*3" is coming in **onMembersAdded** Method
**Description**  

- we are using directline in order to connect to bot service.

-  When the new user is getting added in bot source code it goes for **onMembersAdded** Method. While sending the welcome message to the user  **await context.sendActivity("");** the above error is getting recorded.

Please let me know what is the root cause of this issue. Because of this issue user is not getting any response from the bot.
**Note:** This error is getting recorded in three different places which you can find it in the attachment. Along with this, we are getting a few more errors but we are not aware of whether we can include here or create a new ticket.

DETAILS:
Microsoft AppID: 461494cb-be52-4d6c-af90-2abc2f69dfc2
Timestamp: 2020-07-26T07:25:19.863Z
conversation Id: We don't have
[errorlist.xlsx](https://github.com/microsoft/BotFramework-DirectLineJS/files/4990269/errorlist.xlsx)

**Comment by stevkan** [comment]
Is this a published bot or is it being run in development (i.e. locally)? The error you are receiving indicates that the port is already in use. What is the port you have assigned to the bot?

And, is the error only occurring in the `onMembersAdded` method? If you remove that activity handler and the user interacts with the bot does the error still occur?

Can you verify which version of BotFramework (4.9.2, 4.8.0, etc.)?
Are you able to post your code for the index.js and your main bot file containing the activity handlers?

**Comment by BhargavJanardhan** [comment]
This is a published bot into Azure Bot Service. 

We have enabled auto scaling for the App Service Plan with below configurations.
> Minimum of 1 instance, maximum of 3 instances and by default as 1 instance.
> When the CPU utilization is more than 70% increase by count 1.
> When the CPU utilization is less than 50% decrease by count 1.

![image](https://user-images.githubusercontent.com/27334482/88913966-c955ff00-d259-11ea-8fa2-5a99e9509a86.png)

These errors came mostly on these dates but we also observed the errors came when there is no auto scaling. 
Hence I assume as the bot is deployed in azure and due to auto scale the port number (443) is same. Please confirm.

This error also appeared in Process Activity along with members added. Attached the error list in the above chain.

- **botbuilder node version :** 4.5.3

- **Bot Framework version :** 3.1

- **.Net Framework version :** 4.6

[Code Snippets.zip](https://github.com/microsoft/BotFramework-DirectLineJS/files/5000936/Code.Snippets.zip)

**Comment by stevkan** [comment]
@BhargavJanardhan, a few more questions for you:

- Is this bot written in JS or .Net? Code suggested JS, but above you list ".Net Framework version : 4.6".
- I'm not sure what you mean "botbuilder node version" and "Bot Framework version".
    - Is "Bot Framework" referring to something different than the BotFramework SDK (i.e. "botbuilder node" you mention first)?
    - Why are they set to different versions?
- Are you able to upgrade to the newest version available, v4.9.3 and try again?
- Does this occur when you run it locally?

In the files you included:
- The `Test.js` file is exporting the "QnABot" class, but in `index.js` you are importing "Test" which doesn't exist as a class. Is this a typo or on purpose? And have you tried running while using the correct class?
- Is there a reason you pass your dialog into `Test.js` instead of introducing it in `index.js`? There are multiple ways of achieving the same result, however the [demonstrated method](https://github.com/microsoft/BotBuilder-Samples/blob/master/samples/javascript_nodejs/05.multi-turn-prompt/index.js#L58) is via `index.js`.

**Comment by stevkan** [comment]
Updating for reference: I've been working with Azure Support via email and Teams to try and determine the root cause of the bot producing errors. Initially, thought it might have been a port exhaustion error which lead me to a series of "ReconnectToConversation" calls that were being producted.

After speaking with a Services engineer, we determined that those calls are not the cause, but may be the result of the bot errors. We also determined that the error is in the bot's code which needs to be reviewed while debugging to find when/where the bot is failing.

Waiting for Azure Support to respond.
